Rubberduck.Core/AutoComplete/AutoCompleteBase.cs

```diff
@@ -29,28 +33,69 @@ public virtual bool Execute(AutoCompleteEventArgs e, AutoCompleteSettings settin
             var module = e.CodeModule;
             using (var pane = module.CodePane)
             {
-                var selection = pane.Selection;
-                var original = module.GetLines(selection);
-                var nextChar = selection.StartColumn - 1 == original.Length ? string.Empty : original.Substring(selection.StartColumn - 1, 1);
+                var pSelection = pane.Selection;
+                var zSelection = pSelection.ToZeroBased();
+
+                var original = module.GetLines(pSelection);
+                var nextChar = zSelection.StartColumn == original.Length ? string.Empty : original.Substring(zSelection.StartColumn, 1);
                 if (input == InputToken && (input != OutputToken || nextChar != OutputToken))
                 {
-                    var code = original.Insert(Math.Max(0, selection.StartColumn - 1), InputToken + OutputToken);
-                    module.ReplaceLine(selection.StartLine, code);
-                    pane.Selection = new Selection(selection.StartLine, selection.StartColumn + 1);
+                    var code = original.Insert(Math.Max(0, zSelection.StartColumn), InputToken + OutputToken);
+                    module.ReplaceLine(pSelection.StartLine, code);
+
+                    var newCode = module.GetLines(pSelection);
+                    if (newCode == code)
+                    {
+                        pane.Selection = new Selection(pSelection.StartLine, pSelection.StartColumn + 1);
+                    }
+                    else
+                    {
+                        // VBE added a space; need to compensate:
+                        pane.Selection = new Selection(pSelection.StartLine, GetPrettifiedCaretPosition(pSelection, code, newCode));
```

Comment:
The code is saying to me, "The comment is a lie!" ;-) Either update or remove the comment since we now have the `GetPrettifiedCaretPosition` now.